datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -79,6 +76,193 @@ fn is_not_distinct_from_bool(
         .collect())
 }
 
+// TODO add iter for decimal array
+// TODO move this to arrow-rs
```

Comment:
👍

Reply:
I think it is a good idea to put the kernels into DataFusion first and then port to arrow 👍  There are some tricks to make them faster but that will be good discussions to have in arrow